main/lsp/json_types.cc

```diff
@@ -227,10 +225,10 @@ void LSPFileUpdates::mergeOlder(const LSPFileUpdates &older) {
         }
         encountered.emplace(f->path());
         updatedFiles.push_back(f);
-        updatedFileHashes.push_back(older.updatedFileHashes[i]);
         auto &ast = older.updatedFileIndexes[i];
         updatedFileIndexes.push_back(ast::ParsedFile{ast.tree->deepCopy(), ast.file});
     }
+    canTakeFastPath = false;
```

Comment:
Resets the fast path decision after merging.

I have a follow-up PR to turn this into a trinary value (FAST / SLOW / NOT_DECIDED) so we can crash if we try to typecheck something that we didn't make a fresh fast path decision on.